docx_editor/exceptions.py

```
@@ -141,6 +141,35 @@ def __init__(
+    library already does. Neither does ``w:writeProtection`` (Word's "Password
+    to modify" and "Always Open Read-Only"), which is a different element and
+    out of scope: it restricts saving over the original rather than editing the
+    body.
+
+    Attributes:
+        path: The document that is protected, or None if unknown.
+        mode: The raw ``w:edit`` value that triggered the guard
+            (``"readOnly"``, ``"forms"`` or ``"comments"``), or None if unknown.
+    """
+
+    def __init__(self, message: str, *, path: Path | None = None, mode: str | None = None):
```

Ruff (0.16.2)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)